storage/innobase/xtrabackup/src/ds_decrypt.c

```diff
@@ -358,19 +358,23 @@ decrypt_write(ds_file_t *file, const void *buf, size_t len)
 		thd = threads;
 
 		pthread_mutex_lock(&thd->ctrl_mutex);
+		parse_result = thd->parse_result;
```

Comment:
This line looks wrong to me. We should not pick random `parse_result` here. It actually shouldn't matter what `parse_result` was at the end of previous call of this function.

Reply:
Sure, fixed that nonsense.